src/test/java/org/jabref/logic/importer/ParserResultTest.java

```diff
@@ -48,4 +55,80 @@ public void doesNotHaveDuplicateWarnings() {
         parserResult.addWarning("Duplicate Warning");
         assertEquals("Duplicate Warning", parserResult.getErrorMessage());
     }
+
+    @Test
+    public void warningAddedForWhitespaceInCitationKeyImport(@TempDir Path tmpDir) throws IOException {
+        // whitespace after citation key "myArticle "
+        String bibtexEntry = """
+                @article{ myArticle ,
+                  author    = "Author Name",
+                  title     = "Title of the Article",
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+        File tempFile = new File(tmpDir.toFile(), "invalidBibTex.bib");
```

Comment:
I fixed this in the follow up PR but you can just do:
        Path tempFile = tmpDir.resolve("invalidBibTex.bib");